src/components/PetrovSD/Leader.js

```diff
@@ -0,0 +1,83 @@
+import React, {useState, useEffect} from 'react'
+
+const fetchData = async (props,link,stateUpdate) => {
+    fetch(link)
+    .then(response => response.json())
+    .then(json => {
+        stateUpdate(
+            {
+                isLoading: false,
+                rank: props.rank,
+                player: json.data.names.international,
+                igt: getTime(props.igt),
+                platform: props.platform,
+                date: props.date,
+                link: props.link,
+                loadLink: props.loadLink
+            }
+        )
+    })
+}
+
+const getService = (link) => {
+    if (link.includes('twitch')) {
+        return "Twitch"
+    }
+    if (link.includes('youtu.be') || link.includes('youtube.com')) {
+        return "Youtube"
+    }
+    return "Watch"
+}
+
+const getTime = (time) => {
+    const mins = (time - (time % 60))/60
+    const secs = time - mins*60
+    return mins + "m " + secs + "sec"
+}
+
+const Leader = (props) => {
+    const [ state,setState ] = useState({
+        isLoading: true,
+        rank: null,
+        player: null,
+        igt: null,
+        platform: null,
+        date: null,
+        link: null,
+        loadLink: props.loadLink
+    })
+
+    useEffect(()=>{
+        fetchData(props,props.loadLink,setState)
+    })
+
+    if (state.isLoading === false) {
+        return (
+            <tr key={props.key} className="leader">
+                <th>{state.rank}</th>
+                <th>{state.player}</th>
+                <th>{state.igt}</th>
+                <th>{state.platform}</th>
+                <th>{state.date}</th>
+                <th>
+                    <a href={state.link} target="_blank">
```

Comment:
Имеет потенциальную дыру в безопасности (warning react/jsx-no-target-blank)
[Подробнее](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-target-blank.md)

Следующие изменения исправят проблему:
```suggestion
                    <a href={state.link} target="_blank" rel='noopener noreferrer'>
```

Reply:
fixed